### PR TITLE
Log 11792 journal toggle

### PR DESCRIPTION
**Pull request by james-t-bloom**
bug(journal_toggle):  Wrap task spawning in more conditions including new env variable 

Summary: added new environment variable MZ_SYSTEMD_JOURNAL_TAILER
If user explicitly sets that to false, no matter if there is a journal path no tasks will spawn

Test plan:

Manually testing 
unit tests 
Int tests

Ref: LOG-11792

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -294,7 +295,7 @@ file in docs directory of this repository.
 
 ### Configuring Journald
 
-If the agent pods have access to journald log files or directories, monitoring can be enabled on them with the `LOGDNA_JOURNALD_PATHS`. Common values include `/var/log/journal` and `/run/systemd/journal`. To specify both, use a comma separated list: `/var/log/journal,/run/systemd/journal`.
+If the agent pods have access to journald log files or directories, monitoring can be enabled on them with the `LOGDNA_JOURNALD_PATHS` as well as setting `MZ_SYSTEMD_JOURNAL_TAILER` to true, or leaving blank which defaults to true. Common values include `/var/log/journal` and `/run/systemd/journal`. To specify both, use a comma separated list: `/var/log/journal,/run/systemd/journal`.
```
Comment (dkhokhlov) [inline docs/README.md:298]: How many vars need to be defined to enable journal log? 

How many vars need to be defined to have journal log disabled?

Comment (dkhokhlov) [reply docs/README.md:298]: table needs to be updated:
https://github.com/logdna/logdna-agent-v2/blob/master/docs/README.md#configuration

Comment (james-t-bloom) [reply docs/README.md]: updated a bit

File: Jenkinsfile [hunk 2]
```diff
@@ -65,20 +65,6 @@ pipeline {
         }
         stage('Lint and Unit Test') {
             parallel {
-                stage('Lint'){
```
Comment (dkhokhlov) [inline Jenkinsfile:68]: why removed?

File: bin/src/_main.rs [hunk 3]
```diff
@@ -286,29 +286,44 @@ pub async fn _main(
...
+                        info!("Journalctl source was not initialized");
+                        debug!("Journalctl source initialization error: {}", e);
+                    })
+                    .ok()
+            } else {
+                None
+            }
+        }
+        false => None,
     };
```
Comment (dkhokhlov) [inline bin/src/_main.rs:317]: How do we know journlad is started/initialized?

Comment (james-t-bloom) [reply bin/src/_main.rs:317]: there are two logs in the code one for journalCTL and for ID a little deeper.

File: common/config/src/properties.rs [hunk 4]
```diff
@@ -256,6 +257,11 @@ fn from_property_map(map: HashMap<String, String>) -> Result<Config, ConfigError
             .for_each(|v| paths.push(PathBuf::from(v)));
     }
 
+    if let Some(value) = map.get(&SYSTEMD_JOURNAL_TAILER) {
+        log::error!("HEYYY {}", value);
```
Comment (dkhokhlov) [inline common/config/src/properties.rs:261]: Error message?

**Review by dkhokhlov — CHANGES_REQUESTED** [review]

File: common/config/src/properties.rs (hunk 4 above)
Comment (james-t-bloom) [reply common/config/src/properties.rs:261]: lol 🙄

File: bin/tests/it/cli.rs [hunk 5]
```diff
@@ -610,8 +610,9 @@ async fn test_z_journald_support() {
     settings.journald_dirs = Some(dir);
     settings.features = Some("libjournald");
     settings.exclusion_regex = Some(r"^(?!/var/log/journal).*$");
-    assert_eq!(systemd::journal::print(6, "Sample info"), 0);
+    settings.log_journal_d = Some("true");
```
Comment (dkhokhlov) [inline bin/tests/it/cli.rs:613]: everywhere:
log_journal_d -> log_journald_enabled or log_journald
?

File: docs/README.md [hunk 6]
```diff
@@ -294,7 +295,7 @@ file in docs directory of this repository.
 
 ### Configuring Journald
 
-If the agent pods have access to journald log files or directories, monitoring can be enabled on them with the `LOGDNA_JOURNALD_PATHS`. Common values include `/var/log/journal` and `/run/systemd/journal`. To specify both, use a comma separated list: `/var/log/journal,/run/systemd/journal`.
+If the agent pods have access to journald log files or directories, monitoring can be enabled on them with the `LOGDNA_JOURNALD_PATHS` as well as setting `MZ_SYSTEMD_JOURNAL_TAILER` to true, or leaving blank which defaults to true. Both variables are required to be configured. Common values include for `LOGDNA_JOURNALD_PATHS` - `/var/log/journal` and `/run/systemd/journal`. To specify both, use a comma separated list: `/var/log/journal,/run/systemd/journal`.
```
Comment (stsantilena) [inline docs/README.md:298]: Nit: the wording is a bit confusing. Do they both default to true? So, to enable, you need to either set them BOTH to to true, or leave them BOTH blank?

File: bin/src/_main.rs [hunk 7]
```diff
@@ -286,29 +286,44 @@ pub async fn _main(
...
+            if !config.journald.paths.is_empty() {
+                Some(create_source(&config.journald.paths).map(StrictOrLazyLineBuilder::Strict))
+            } else {
+                None
+            }
+        }
+        false => None,
+    };
+
+    #[cfg(all(feature = "libjournald", target_os = "linux"))]
```
Comment (stsantilena) [inline bin/src/_main.rs:301]: Nit: These annotations can get confusing, especially when the naming within this section so is similar. Perhaps add a small comment just to clarify?

**Review by stsantilena — APPROVED** [review]
Just a couple minor things. LGTM.

**Pull request merged**